android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java

```diff
@@ -1726,8 +1726,14 @@ protected void registerForTouch(final View touchable)
 				touchable.setEnabled(false);
 			}
 		}
+		//Checking and setting touch sound for view
+		if (proxy.hasProperty(TiC.PROPERTY_TOUCH_SOUND_ENABLED)) {
+		 boolean soundEnabled = TiConvert.toBoolean(proxy.getProperty(TiC.PROPERTY_TOUCH_SOUND_ENABLED), true);
+			if (!soundEnabled) {
```

Comment:
If this is `false` first, setting it to `true` will skip every time. Rather call the setter directly.